### PR TITLE
[ClamAV] Adds PVC and imagePullSecrets

### PR DESCRIPTION
**Pull request by zedtux**
This PR brings 5 new things:

* setting the `imagePullSecrets` avoiding to hit docker servers limitations on downloading images
* setting extra volumes in order to make clamav pod running with the `readOnlyRootFilesystem` as `true` for instance
* choosing to use an `emptyDir` or a `PVC` when mounting the `/data` folder
* converting the `Deployment` manifest to a `StatefulSet` one
* Fixes `resources` which was causing a lint offense when passing `limits` and `requests`

### CODE REVIEW

**Comment by desaintmartin** [comment]
Thanks for the change! I am not comfortable having Helm managing PVCs, should we simply change this to a statefulset?

**Comment by zedtux** [comment]
Well, I didn't wanted to change too much things to this Helm chart ... for now as I'm preparing another PR to be able to choose between a `Deployment` or a `DaemonSet`.

Nonetheless it would make much more sense to me too to use a `StatefulSet` instead, so let's do it!

Merci pour ton message 😉

**Comment by zedtux** [comment]
Hello @desaintmartin,

Here you are with the `StatefulSet`. I've tested it, and it work well.

File: charts/clamav/templates/statefullset.yaml [hunk 1]
```diff
@@ -56,27 +61,59 @@ spec:
...
+        {{- if .Values.clamdConfig }}
+        - name: clamd-config-volume
+          configMap:
+            name: {{ include "clamav.fullname" . }}-clamd
+        {{- end }}
+        {{- if .Values.extraVolumes }}
+        {{- toYaml .Values.extraVolumes | nindent 8 }}
+        {{- end }}
+  {{- if not .Values.persistentVolume.enabled }}
+        - name: redis-data
```
Comment (desaintmartin) [inline charts/clamav/templates/statefullset.yaml:92]: Redis ;)

**Review by desaintmartin — CHANGES_REQUESTED** [review]

File: charts/clamav/templates/statefullset.yaml (hunk 1 above)
Comment (zedtux) [reply charts/clamav/templates/statefullset.yaml:92]: Damned ... :)

Comment (zedtux) [reply charts/clamav/templates/statefullset.yaml]: Fixed!

**Comment by desaintmartin** [comment]
It seems you need to bump version (this anti-mining feature on github is very annoying, I have to manually authorize test each time...)

**Comment by zedtux** [comment]
> It seems you need to bump version

Oh really? That definitely doesn't sound good to me. I'm contributing, you're creating the release.
What if there's another contributor and you'd like to release both contribution together within the same version?

**Comment by desaintmartin** [comment]
This is actually a "good practice" within the helm community: each merged PR results in a new released version. See the artifact hub for examples! One could argue about the cons and pros, of course.

**Comment by zedtux** [comment]
Then I will update it tomorrow.

**Comment by desaintmartin** [comment]
Thanks!

**Comment by zedtux** [comment]
Can you please let me know when this version will be available on the repo?

**Comment by desaintmartin** [comment]
CD already released it! Artifact Hub may take ~15 minutes to reload its cache, but `helm repo up` will show it already.

**Pull request merged**